src/app/crud/quality_matrix.py

```diff
@@ -1,31 +1,20 @@
-import json
-
 from elasticsearch_dsl import AttrDict
 from elasticsearch_dsl.response import Response
 
-from app.core.constants import (
-    PROPERTIES,
-    REPLICATION_SOURCE,
-    REPLICATION_SOURCE_ID,
-    TOTAL_COUNT,
-)
+from app.core.constants import PROPERTIES, REPLICATION_SOURCE_ID
 from app.core.logging import logger
 from app.crud.elastic import base_match_filter
 from app.elastic import Search, qbool, qmatch
 
+PROPERTY_TYPE = list[str]
```

Comment:
I would just inline this - seems like an unnecessary indirection to me :stuck_out_tongue:

Reply:
The advantage here is, all properties returned have this type - this is a constant, which is more telling than just list[str]. Or do I misunderstand the "inline this"?

Reply:
No, that was exactly what I meant with inline :)

It really is just a personal preference I guess. 

Having it inlined tells the reader that its a list of strings without having to look it up. IMHO, that's good as long as the types are simple enough. Onces the types get more complicated (like `dict[str,Union[str,list[Union[str,float]]]` I always feel that either the function is to complicated, or I should have a dedicated dataclass for the return value.

Having a dedicated type alias (like the `PROPERTY_TYPE = list[str]`) becomes usefull if there are special guarantees attached with the list and the strings (e.g. list is never empty, items in list can only have a certain set of values). And even more so when there are multiple `list[str]` functions with different guarantees about their return values.

Reply:
I agree with the dedicated data classes. I like the explicit name, because it contextualizes the "list[str]". So baiscally, I know it is a list of properties in the shape of strings. A data class is more rigid, but also more explicit.

I also like the point of guarantee, the way you phrase it it sounds like another layer of explicitness, right? I will keep it in mind. :) Thanks for pointing it out!